In [2]:
import numpy as np
import pylab as pl
import pandas as pd
pl.style.use('fivethirtyeight')

In [3]:
%pylab inline

Populating the interactive namespace from numpy and matplotlib


In [4]:
#loading the data
mta = np.load('MTA_Fare.npy')
type = np.array(['10t', '14d', '1d', '2t', '30d', '7d', 'ada', 'afas', 'exp','ez',
            'ff','mc','mr','month','rfm','rr','sen','spec','tcmc','t',
            'umlim','xbus','stud'])
mta = np.where(mta<0, 'Nan', mta)
mta = mta.astype(float)
time = pd.date_range('05/21/2010', periods=194, freq='w')

In [5]:
#getting the mean of all the stations
x = np.nanmean(mta, axis=1)

//anaconda/lib/python2.7/site-packages/numpy/lib/nanfunctions.py:598: RuntimeWarning: Mean of empty slice
  warnings.warn("Mean of empty slice", RuntimeWarning)


In [6]:
x

array([[ 5767.18181818,  6794.40909091,  6926.5       , ...,
         3430.47826087,  3243.08695652,  3340.56521739],
       [ 1695.72727273,  2112.        ,  2243.59090909, ...,
                   nan,            nan,            nan],
       [  300.27272727,   312.77272727,   328.59090909, ...,
          355.26086957,   343.7826087 ,   295.2173913 ],
       ..., 
       [           nan,            nan,            nan, ...,
                   nan,            nan,            nan],
       [           nan,            nan,            nan, ...,
                   nan,            nan,            nan],
       [           nan,            nan,            nan, ...,
                   nan,            nan,            nan]])

In [7]:
#a little formula that will take the max of each station and the subtract it from the sd. 
score = []
for i in range(len(x)):
    trans = np.abs(np.fft.rfft(x[i]))
    score.append(np.max(trans)-np.std(trans))
    
#printing the stations with the most highest periodic trends. 
scframe = pd.DataFrame(score, columns=['score'])
scframe.sort(['score'], ascending=False)[:4]

score
169  4852193.848318
21   3121754.221772
45   3044248.445528
83   3032394.758747

The four stations witht he highest periodicity are 169, 21, 45, and 83. 